importlib_metadata/__init__.py

```diff
@@ -159,6 +156,9 @@ class EntryPoint(
 
     dist: Optional['Distribution'] = None
 
+    def __init__(self, *, name, value, group):
```

Comment:
I would recommend going for explicit assignment here for the sake of readability and setting self.dist as well

Reply:
I'd actually like to keep the name/group/value as the essential attributes separate from the dist, which is optional and auxiliary. Agree with explicit assignment. I'm not sure it's possible.